hermes/probe/read/read.go

```diff
@@ -0,0 +1,124 @@
+	"google.golang.org/api/iterator"
+)
+
+const (
+	FileNameFormat          = "Hermes_%02d_%x"
+	minFileID               = 1
+	maxFileID               = 50
+	maxFileSizeBytes        = 1000
+	hermesAPILatencySeconds = "hermes_api_latency_s"
+)
+
+// ReadFile reads a file with randomized contents in the target storage system.
+// It verifies the availability and consistency of the file's contents.
+// Finally, it records the exit status in the logger.
+// Arguments:
+//          ctx: it carries deadlines and cancellation signals that might orinate from the main probe located in probe.go.
+//          target: contains information about target storage system, carries an intent log in the form of a StateJournal and it used to export metrics.
+//          fileID: the unique identifer of every file, it cannot be repeated. It needs to be in the range [minFileID, maxFileID]. FileID 0 is reserved for a special file called the NIL file.
+//          client: is a storage client. It is used as an interface to interact with the target storage system.
+//          logger: a cloudprober logger used to record the exit status of the CreateFile operation in a target bucket. The logger passed MUST be a valid logger.
+// Returns:
+//          error: an error string with detailed information about the status and fileID. Nil is returned when the operation is successful.
+func ReadFile(ctx context.Context, target *probe.Target, fileID int32, fileSize int, client stiface.Client, logger *logger.Logger) error {
+	if fileID < minFileID || fileID > maxFileID {
+		return fmt.Errorf("invalid argument: fileID = %d; want %d <= fileID <= %d", fileID, minFileID, maxFileID)
+	}
+	bucketName := target.Target.GetBucketName()
+
+	// Verify that the file is present in the State Journal
+	fileName, ok := target.Journal.Filenames[fileID]
+	if !ok {
+		return fmt.Errorf("file with the ID: %d is missing from the State Journal", fileID)
+	}
+
+	// Verify that the file that we want to read is in fact present in the target system
+	fileNamePrefix := fmt.Sprintf(FileNameFormat, fileID, "")
+	query := &storage.Query{Prefix: fileNamePrefix}
+	start := time.Now()
+	objIter := client.Bucket(bucketName).Objects(ctx, query)
+	var namesFound []string
+	for {
+		obj, err := objIter.Next()
+		if err == iterator.Done {
+			break
+		}
+		if err != nil {
+			return fmt.Errorf("ReadFile(ID: %d) check failed due to: %w", fileID, err)
+		}
+		namesFound = append(namesFound, obj.Name)
+	}
+	finish := time.Now()
+	if len(namesFound) == 0 {
+		target.LatencyMetrics.APICallLatency[metrics.APIListFiles][metrics.FileMissing].Metric(hermesAPILatencySeconds).AddFloat64(finish.Sub(start).Seconds())
+		return fmt.Errorf("ReadFile(ID: %d) could not read file as the file with the provided ID does not exist on the target system", fileID)
+	}
+	if len(namesFound) != 1 {
+		return fmt.Errorf("expected exactly one file in bucket %q with prefix %q; found %d: %v", bucketName, fileNamePrefix, len(namesFound), namesFound)
```

Comment:
Keep the error messages consistent: either put the function name in all of them or none of them.
I would lean towards not putting the function name there, as it should be the responsibility of the caller to include it (if it's relevant).

Reply:
removed function name from errors added in in read_test.go errors starting from a small letter